### ISSUE TITLE
Typing lag experienced in larger Obs. vault (iPadOS 15.6)

### ISSUE THREAD

**Issue by zanodor** [TfTHacker/obsidian42-strange-new-worlds#43]
**Re: Our discussion on Obsidian Forum**

Vault size 14200 plus 4000 images.

Maybe a setting to allow indexing once every few hours or even a day would solve it? 

No idea.

Thanks for the great plugin and your dedication.

**Comment by TfTHacker** [comment]
I assume there isn't a chance of getting my hands on the vault for testing purposes? (I'll sign an NDA 😄 )

**Comment by TfTHacker** [comment]
Tnx! was just going to download. I am curious - are you seeing this problem just on the iPad or also on desktop.

I will be testing this with my MacBook Air M1 - which likely won't be slow. But can test on my 2018 iPad Pro, which for sure runs slower.

The larger the vault, the more hardware it wants, which I think is reasonable.

**Comment by alltagsverstand** [comment]
I can confirm this issue - even on my very fast PC obsidian was hanging repeatedly when typing (though my vault "only" contains around 5.000 files, most of them (about 95%) being just simple md files). After disabling the plugin, everything went smooth again.

**Comment by TfTHacker** [comment]
@alltagsverstand thank you for your report. Is your vault heavily linked? Also, I assume vault is private, but if its not, I would love to do testing with it.

**Comment by alltagsverstand** [comment]
In fact, my vault contains a lot of very private stuff as I am collecting everything within one single large vault.

Hmm, what would "heavily" linked mean? According to the vault statistics plugin my vault includes 7.680 links...

**Comment by TfTHacker** [comment]
@alltagsverstand would you mind doing another test? could you disable other plugins and try just SNW? I don't think your link count is a problem. just curious if maybe there is another plugin being triggered by something SNW is doing.

**Comment by TfTHacker** [comment]
@zanodor I have done some testing with your vault. It is large 😄 .  18,3000 files and 156,000+ links. Impressive!

Few observations: I think this vault on the iPad min 4 will be slow with SNW, SNW does a lot of calculations when you open a file. I tried it on my iPad Pro 2018, and while the vault is not fast, it was usable. On my M1 it is good.

Though I did notice, Obsidian even when doing [[ ]] lookups of files is a bit sluggish (with no plugins enabled). 

I am going to continue to investigate performance on the plugin, but SNW for sure is going to require more CPU horse power. One thing you can do is use settings to disable some SNW features. For example there is a gutter setting which in the end might not be a big deal you. Or maybe you don't link to headings frequently, so you can turn the counter off for that.

In other words, you can get some benefit from SNW, with some features turned off.

**Comment by alltagsverstand** [comment]
> @alltagsverstand would you mind doing another test? could you disable other plugins and try just SNW? I don't think your link count is a problem. just curious if maybe there is another plugin being triggered by something SNW is doing.

Strange enough, after reactivating the plugin everything seems to work fine for the moment... I will keep an eye on that and report/do some tests in case the problem will occur again!

**Comment by TfTHacker** [comment]
Regarding size of the numbers, I made them bigger since people were having a hard time clicking on them if they are small on mobile. Having said that, if you are comfortable with CSS you can easily modify the size of the numbers. More info here:

https://github.com/TfTHacker/obsidian42-strange-new-worlds/discussions/19

Regarding "daily update of links" - I didn't understand what you meant by that. Can you give me some more details?

**Comment by l0wfl0w** [comment]
I can also confirm lagging while typing on iPad Air 4th Gen (2020) with a vault of ~2900 notes and probably heavy linking (can’t share the vault sorry). I realized it gets mich worse when also turning on tasks plugin in parallel.
Will check in some days if same behavior on Mac Mini m1 , assume it will be better there.

(besides that: thanks for this amazing plugin! For me it adds the *only* feature I really missed from Roam! Assume this is a strong candidate to be sherlocked at some point in future?😇)
